akka-actor/src/main/scala/akka/pattern/AskSupport.scala

```diff
@@ -328,7 +328,9 @@ private[akka] object PromiseActorRef {
   private case object Stopped
   private final case class StoppedWithPath(path: ActorPath)
 
-  def apply(provider: ActorRefProvider, timeout: Timeout, targetName: String, message: Any, sender: ActorRef = Actor.noSender): PromiseActorRef = {
+  private val ActorStopResult = Failure(new ActorKilledException("Stopped"))
+
+  def apply(provider: ActorRefProvider, timeout: Timeout, targetName: ⇒ String, message: Any, sender: ActorRef = Actor.noSender): PromiseActorRef = {
```

Comment:
All callsites use ActorRef#toString as argument which in turn uses ActorRef#path and path#id.

Every implementation of path i've found is an val except for PromiseActorRef's. 
path.id is a alwyas val or ActorCell.undefinedUid.
PromiseActorRef#path on the other hand is creating a path unless it already is one, in seemingly threadsafe manners.